roxie/udplib/udpaeron.cpp

```diff
@@ -0,0 +1,418 @@
+        {
+            SpinBlock b(collatorsLock);
+            try
+            {
+                msgColl.set(collators[pktHdr->ruid]);
+                if (!msgColl)
+                {
+                    msgColl.set(collators[RUID_DISCARD]);
+                    // We could consider sending an abort to the slave, but it should have already been done by ccdserver code
+                    isDefault = true;
+                    unwantedDiscarded++;
+                }
+            }
+            catch (IException *E)
+            {
+                EXCLOG(E);
+                E->Release();
+            }
+            catch (...)
+            {
+                IException *E = MakeStringException(ROXIE_INTERNAL_ERROR, "Unexpected exception caught in CPacketCollator::run");
+                EXCLOG(E);
+                E->Release();
+            }
+        }
+        if (udpTraceLevel && isDefault)
+        {
+            StringBuffer s;
+            DBGLOG("AeronReceiver: CPacketCollator NO msg collator found - using default - ruid=" RUIDF " id=0x%.8X mseq=%u pkseq=0x%.8X node=%s", pktHdr->ruid, pktHdr->msgId, pktHdr->msgSeq, pktHdr->pktSeq, pktHdr->node.getTraceText(s).str());
+        }
+        if (msgColl)
+            msgColl->attach_data(buffer, length);
+    }
+
+    // Note - some of this code could be in a common base class with udpreceivemanager, but hope to kill that at some point
+    virtual IMessageCollator *createMessageCollator(roxiemem::IRowManager *rowManager, ruid_t ruid) override
+    {
+        CMessageCollator *msgColl = new CMessageCollator(rowManager, ruid);
+        if (udpTraceLevel >= 2)
+            DBGLOG("AeronReceiver: createMessageCollator %p %u", msgColl, ruid);
+        {
+            SpinBlock b(collatorsLock);
+            collators[ruid] = msgColl;
+        }
+        msgColl->Link();
+        return msgColl;
+    }
+
+    virtual void detachCollator(const IMessageCollator *msgColl) override
+    {
+        ruid_t ruid = msgColl->queryRUID();
+        if (udpTraceLevel >= 2)
+            DBGLOG("AeronReceiver: detach %p %u", msgColl, ruid);
+        {
+            SpinBlock b(collatorsLock);
+            collators.erase(ruid);
+        }
+        msgColl->Release();
+    }
+
+private:
+
+    std::shared_ptr<aeron::Subscription> addSubscription(const SocketEndpoint &myEndpoint, int queue)
+    {
+        StringBuffer channel("aeron:udp?endpoint=");
+        myEndpoint.getIpText(channel);
+        channel.append(':').append(myEndpoint.port);
+        std::int64_t id = aeron->addSubscription(channel.str(), queue);
+        std::shared_ptr<aeron::Subscription> subscription = aeron->findSubscription(id);
+        while (!subscription)
+        {
+            std::this_thread::yield();
+            subscription = aeron->findSubscription(id);
+        }
+        return subscription;
+    }
+};
+
+class UdpAeronReceiverEntry : public IUdpReceiverEntry
```

Comment:
I've suffered from similar confusions throughout the life of the UDP layer. Does a SendManager send data or manage information about a bunch of remote senders?  I've (more or less) maintained existing naming so far, and this is a receiverEntry because the udp version is, but a wholesale renaming may make some sense (if some unambiguous names can be found)